renderlib/CCamera.cpp

```diff
@@ -2,6 +2,81 @@
 
 #include "gesture/gesture.h"
 
+void
+CCamera::SetViewMode(const EViewMode ViewMode)
+{
+  if (ViewMode == ViewModeUser)
+    return;
+
+  glm::vec3 ctr = m_SceneBoundingBox.GetCenter();
+  m_Target = ctr;
+  m_Up = glm::vec3(0.0f, 1.0f, 0.0f);
+
+  const float size = m_SceneBoundingBox.GetDiagonalLength();
+  const float Length = (m_Projection == ORTHOGRAPHIC) ? 2.0f : size * 0.5f / tan(0.5f * m_FovV * DEG_TO_RAD);
+  m_OrthoScale = DEF_ORTHO_SCALE;
+
+  // const float Distance = 0.866f;
+  // const float Length = Distance * m_SceneBoundingBox.GetMaxLength();
+
+  m_From = m_Target;
+
+  switch (ViewMode) {
+    case ViewModeFront:
+      m_From.z += Length;
+      m_Up = glm::vec3(0.0f, 1.0f, 0.0f);
+      break;
+    case ViewModeBack:
+      m_From.z -= Length;
+      m_Up = glm::vec3(0.0f, 1.0f, 0.0f);
+      break;
+    case ViewModeLeft:
+      m_From.x += Length;
+      m_Up = glm::vec3(0.0f, 0.0f, 1.0f);
```

Comment:
When you are looking at side views, I want the top of the volume to be the positive Z direction always.   For z-stack volumes this totally makes sense.  Anything else, it would look sideways.